library/Zend/Http/Client/Adapter/Curl.php

```diff
@@ -421,19 +421,26 @@ public function write($method, $uri, $httpVersion = 1.1, $headers = array(), $bo
             throw new AdapterException\RuntimeException("Error in cURL request: " . curl_error($this->curl));
         }
 
+        // separating header from body because it is dangerous to accidentially replace strings in the body
+        $responseHeaderSize = curl_getinfo($this->curl, CURLINFO_HEADER_SIZE);
+        $responseHeaders = substr($this->response, 0, $responseHeaderSize);
+
         // cURL automatically decodes chunked-messages, this means we have to disallow the Zend\Http\Response to do it again
-        if (stripos($this->response, "Transfer-Encoding: chunked\r\n")) {
-            $this->response = str_ireplace("Transfer-Encoding: chunked\r\n", '', $this->response);
-        }
+        $responseHeaders = preg_replace("/Transfer-Encoding:\s*chunked\\r\\n/", "", $responseHeaders);
 
```

Comment:
In most cases the `Transfer-Encoding:*whitespace*chunked` is true, but I found some cases (at that time back than buzzfeed.com for example) where there was no whitespace between the two parts. Problem: I was not able to send a header without that whitespace. Maybe the server or php corrects the header before sending.